src/Microsoft.AspNetCore.SignalR.Redis/RedisHubLifetimeManager.cs

```diff
@@ -80,34 +81,66 @@ public class RedisHubLifetimeManager<THub> : HubLifetimeManager<THub>, IDisposab
 
                 previousBroadcastTask = Task.WhenAll(tasks);
             });
+
+            channelName = _channelNamePrefix + ".AllExcept";
+            _logger.LogInformation("Subscribing to channel: {channel}", channelName);
+            _bus.Subscribe(channelName, async (c, data) =>
+            {
+                await previousBroadcastTask;
+
+                _logger.LogTrace("Received message from redis channel {channel}", channelName);
+
+                var message = DeserializeMessage<RedisExcludeClientsMessage>(data);
+                var excludedIds = message.ExcludedIds;
+
+                //// TODO: This isn't going to work when we allow JsonSerializer customization or add Protobuf
+
+                var tasks = new List<Task>(_connections.Count);
+
+                foreach (var connection in _connections)
+                {
+                    if (!excludedIds.Contains(connection.ConnectionId))
+                    {
+                        tasks.Add(WriteAsync(connection, message));
+                    }
+                }
+
+                previousBroadcastTask = Task.WhenAll(tasks);
```

Comment:
I think this is wrong/racey - you can replace a task that has not been finished with another task that is not finished. Sharing the same task between broadcast and .allexcept makes it even more racey